src/components/ui/Rotation.tsx

```diff
@@ -2,13 +2,14 @@ import cn from 'classnames'
 import {ActionLink, ItemLink} from 'components/ui/DbLink'
 import {getDataBy} from 'data'
 import ACTIONS, {ITEM_ID_OFFSET} from 'data/ACTIONS'
+import {Cause} from 'event'
 import {Ability} from 'fflogs'
 import React, {Component} from 'react'
 import overlayStyle from './Procs/ProcOverlay.module.css'
 import styles from './Rotation.module.css'
 
 interface RotationProps {
-	events: Array<{ability: Ability, isProc?: boolean}>
+	events: Array<{ability?: Ability, cause?: Cause, action?: number, isProc?: boolean}>
```

Comment:
While we're touching this anyway... can we extract this object as an interface that can then be shared with RotationTable, so we don't have to keep editing the two in tandem?

Reply:
Sure, that makes sense. Done.